src/SmartComponents/SystemAdvisor/SystemAdvisor.js

```diff
@@ -185,14 +185,9 @@ const BaseSystemAdvisor = () => {
         (rowVal, rowKey) =>
           rowKey % 2 === 0 && rowVal.rule.rule_id === rule.rule_id && rowVal
       );
-      const selected = entity.length ? entity[0].selected : false;
-      const isOpen = rows.length
-        ? entity.length
-          ? entity[0].isOpen
-          : false
-        : key === 0
-        ? true
-        : false;
+
+      const selected = (entity.length && entity[0].selected) || false;
+      const isOpen = (entity.length &&  entity[0].isOpen) || false;
```

Comment:
Won't `.length` always evaluate to true, since `.filter` will always return an array even if it has no items?

Reply:
Maybe just `entity[0]?.selected` works?

Reply:
I believe **.length** returns **0** value when the array is empty which evaluates to false in condition evaluation. 

`const arra= [];
const secondarr = [1];
console.log(arra.length && 'hello' || 'bye', secondarr.length);`  => will output 'bye', 1

Reply:
> I believe .length returns 0 value when the array is empty which evaluates to false in condition evaluation.

Sorry, yeah, @johnsonm325 pointed it out to me as well. Totally forgot about that 0 === false.

Reply:
Actullay if I remove the .length, then the page may break becuase of  an empty entity array (**entity[0].isOpen** and **entity[0].selected**) filtered out on line [#184](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/SmartComponents/SystemAdvisor/SystemAdvisor.js#L184) We need to check that entity exists. 

Actually I would do this totally different, but this needs to be accross the app and requires a lot of change :)

Reply:
this error is what I would face:
![image](https://user-images.githubusercontent.com/59481011/151529380-28bc36c9-a97d-48bb-a373-c950bca15459.png)

Reply:
ACK! Thank you for further investigating and elaborating on it!